### ISSUE TITLE
Empty event is not filtered out from event reaction results in Process Manager in case Either is used

### ISSUE THREAD

**Issue by yuri-sergiichuk** [SpineEventEngine/core-java#692]
Currently, in case `EitherOf` is used as a return type for `@React` method in Process Manager and is populated with `Empty` it will not be filtered out inside `ProcessManager#dispatchEvent` and a list with `empty` event is returned:
```java
List<Event> dispatchEvent(EventEnvelope event)  {
  final EventReactorMethod method = thisClass().getReactor(event.getMessageClass());
  final List<? extends Message> eventMessages =
        method.invoke(this, event.getMessage(), event.getEventContext());
  final List<Event> events = toEvents(eventMessages, event); // <--- we should filter results here
  return events;
}
```

In aggregates, it works, cause there is an additional filtration with a `NonEmpty` predicate:
```java
protected List<? extends Message> dispatchCommand(CommandEnvelope command) {
  idempotencyGuard.check(command);
  final CommandHandlerMethod method = thisClass().getHandler(command.getMessageClass());
  final List<? extends Message> messages =
        method.invoke(this, command.getMessage(), command.getCommandContext());
  return from(messages).filter(nonEmpty()).toList();
}
```

**Comment by mdrachuk** [comment]
Implemented in https://github.com/SpineEventEngine/core-java/pull/697.

**Comment by mdrachuk** [comment]
Implemented in #697.
